### ISSUE TITLE
[BUG] Not processing documents and huge memory consumption

### ISSUE THREAD

**Issue by sachilles** [jonaswinkler/paperless-ng#537]
Since version 1.1.0 I have the problem that no more documents are consumed. In particular, it is very noticeable that the RAM used by paperless-ng almost epxlodes and sometimes fills more than 29 GB.

In addition, I get error messages of the type,

```
[2021-02-13 17:53:21,285] [INFO] [paperless.consumer] Consuming ct.21.05.pdf
[2021-02-13 17:56:56,356] [WARNING] [ocrmypdf._pipeline] Some input metadata could not be copied because it is not permitted in PDF/A. You may wish to examine the output PDF's XMP metadata.
17:59:52 [Q] INFO Enqueued 1
17:59:52 [Q] INFO Process-1 created a task from schedule [Optimize the index]
17:59:52 [Q] INFO Process-1:2 processing [michigan-king-spring-fillet]
18:01:01 [Q] ERROR Failed [ct.21.05.pdf] - Error 104 while writing to socket. Connection reset by peer. : Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/django_redis/cache.py", line 27, in _decorator
    return method(self, *args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/django_redis/cache.py", line 76, in set
    return self.client.set(*args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/django_redis/client/default.py", line 166, in set
    raise ConnectionInterrupted(connection=client) from e
django_redis.exceptions.ConnectionInterrupted: Redis ConnectionError: Error 104 while writing to socket. Connection reset by peer.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/django_q/cluster.py", line 436, in worker
    res = f(*task["args"], **task["kwargs"])
  File "/usr/src/paperless/src/documents/tasks.py", line 82, in consume_file
    task_id=task_id
  File "/usr/src/paperless/src/documents/consumer.py", line 267, in try_consume_file
    classifier = load_classifier()
  File "/usr/src/paperless/src/documents/classifier.py", line 43, in load_classifier
    version=version, timeout=86400)
  File "/usr/local/lib/python3.7/site-packages/django_redis/cache.py", line 34, in _decorator
    raise e.__cause__
  File "/usr/local/lib/python3.7/site-packages/django_redis/client/default.py", line 156, in set
    return bool(client.set(nkey, nvalue, nx=nx, px=timeout, xx=xx))
  File "/usr/local/lib/python3.7/site-packages/redis/client.py", line 1801, in set
    return self.execute_command('SET', *pieces)
  File "/usr/local/lib/python3.7/site-packages/redis/client.py", line 900, in execute_command
    conn.send_command(*args)
  File "/usr/local/lib/python3.7/site-packages/redis/connection.py", line 726, in send_command
    check_health=kwargs.get('check_health', True))
  File "/usr/local/lib/python3.7/site-packages/redis/connection.py", line 718, in send_packed_command
    (errno, errmsg))
redis.exceptions.ConnectionError: Error 104 while writing to socket. Connection reset by peer.

18:10:20 [Q] INFO Processed [michigan-king-spring-fillet]
18:12:25 [Q] INFO Enqueued 1
18:12:25 [Q] INFO Process-1 created a task from schedule [Train the classifier]
18:12:25 [Q] INFO Process-1:1 processing [william-lemon-skylark-queen]
18:13:14 [Q] ERROR Failed [william-lemon-skylark-queen] - Error 104 while writing to socket. Connection reset by peer. : Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/django_redis/cache.py", line 27, in _decorator
    return method(self, *args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/django_redis/cache.py", line 76, in set
    return self.client.set(*args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/django_redis/client/default.py", line 166, in set
    raise ConnectionInterrupted(connection=client) from e
django_redis.exceptions.ConnectionInterrupted: Redis ConnectionError: Error 104 while writing to socket. Connection reset by peer.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/django_q/cluster.py", line 436, in worker
    res = f(*task["args"], **task["kwargs"])
  File "/usr/src/paperless/src/documents/tasks.py", line 44, in train_classifier
    classifier = load_classifier()
  File "/usr/src/paperless/src/documents/classifier.py", line 43, in load_classifier
    version=version, timeout=86400)
  File "/usr/local/lib/python3.7/site-packages/django_redis/cache.py", line 34, in _decorator
    raise e.__cause__
  File "/usr/local/lib/python3.7/site-packages/django_redis/client/default.py", line 156, in set
    return bool(client.set(nkey, nvalue, nx=nx, px=timeout, xx=xx))
  File "/usr/local/lib/python3.7/site-packages/redis/client.py", line 1801, in set
    return self.execute_command('SET', *pieces)
  File "/usr/local/lib/python3.7/site-packages/redis/client.py", line 900, in execute_command
    conn.send_command(*args)
  File "/usr/local/lib/python3.7/site-packages/redis/connection.py", line 726, in send_command
    check_health=kwargs.get('check_health', True))
  File "/usr/local/lib/python3.7/site-packages/redis/connection.py", line 718, in send_packed_command
    (errno, errmsg))
redis.exceptions.ConnectionError: Error 104 while writing to socket. Connection reset by peer.

19:12:49 [Q] INFO Enqueued 1
19:12:49 [Q] INFO Process-1 created a task from schedule [Train the classifier]
19:12:49 [Q] INFO Process-1:2 processing [venus-mountain-item-bakerloo]
19:13:33 [Q] ERROR Failed [venus-mountain-item-bakerloo] - Error 104 while writing to socket. Connection reset by peer. : Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/django_redis/cache.py", line 27, in _decorator
    return method(self, *args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/django_redis/cache.py", line 76, in set
    return self.client.set(*args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/django_redis/client/default.py", line 166, in set
    raise ConnectionInterrupted(connection=client) from e
django_redis.exceptions.ConnectionInterrupted: Redis ConnectionError: Error 104 while writing to socket. Connection reset by peer.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/django_q/cluster.py", line 436, in worker
    res = f(*task["args"], **task["kwargs"])
  File "/usr/src/paperless/src/documents/tasks.py", line 44, in train_classifier
    classifier = load_classifier()
  File "/usr/src/paperless/src/documents/classifier.py", line 43, in load_classifier
    version=version, timeout=86400)
  File "/usr/local/lib/python3.7/site-packages/django_redis/cache.py", line 34, in _decorator
    raise e.__cause__
  File "/usr/local/lib/python3.7/site-packages/django_redis/client/default.py", line 156, in set
    return bool(client.set(nkey, nvalue, nx=nx, px=timeout, xx=xx))
  File "/usr/local/lib/python3.7/site-packages/redis/client.py", line 1801, in set
    return self.execute_command('SET', *pieces)
  File "/usr/local/lib/python3.7/site-packages/redis/client.py", line 900, in execute_command
    conn.send_command(*args)
  File "/usr/local/lib/python3.7/site-packages/redis/connection.py", line 726, in send_command
    check_health=kwargs.get('check_health', True))
  File "/usr/local/lib/python3.7/site-packages/redis/connection.py", line 718, in send_packed_command
    (errno, errmsg))
redis.exceptions.ConnectionError: Error 104 while writing to socket. Connection reset by peer.

```

which suggest that the problem occurs in conjunction with Redis.

**To Reproduce**
1. Start paperless-ng docker services
2. Let a document consume
3. Wait a couple of minutes
4. Check docker logs and monitor the memory consumption

**Expected behavior**
Consumption of the corresponding document without porblems or extremely high memory consumption.

**Screenshots**
<img width="768" alt="Bildschirmfoto 2021-02-13 um 19 28 30" src="https://user-images.githubusercontent.com/4097457/107857954-c7d4d880-6e31-11eb-8a16-2a219a516a62.png">

Memory usage after just about 90 minutes.

**Relevant information**
 - Archlinux
 - Installation method: docker
 - Docker version 20.10.3, build 48d30b5b32
 - Docker image: jonaswinkler/paperless-ng: 1.1.0 (or 1.1.1)
 
**docker-compose.yml**

```
version: "3.4"
services:

  redis:
    image: redis:6.0
    restart: always
    volumes:
      - /etc/localtime:/etc/localtime:ro
      - ./redis/config:/usr/local/etc/redis
    command: [ "redis-server", "/usr/local/etc/redis/redis.conf" ]

  gotenberg:
    image: thecodingmachine/gotenberg
    environment:
        DISABLE_GOOGLE_CHROME: 1

  tika:
    image: apache/tika

  paperless:
    image: jonaswinkler/paperless-ng:1.1.1
    restart: always
    depends_on:
      - redis
      - tika
      - gotenberg
    links:
      - "redis:broker"
    ports:
      - 127.0.0.1:3020:8000
    healthcheck:
      test: ["CMD", "curl", "-f", "http://localhost:8000"]
      interval: 30s
      timeout: 10s
      retries: 5
    volumes:
      - /home/www-data/paperless-ng/data:/usr/src/paperless/data
      - /home/www-data/paperless-ng/media:/usr/src/paperless/media
      - /home/www-data/paperless-ng/export:/usr/src/paperless/export
      - /home/www-data/paperless-ng/consume:/usr/src/paperless/consume
    env_file:
      - .env_paperless
      - .env_paperless-ng
```

**./redis/config/redis.conf**

```
bind 0.0.0.0
protected-mode yes
port 6379
tcp-backlog 511
timeout 0
tcp-keepalive 300
daemonize no
supervised no
pidfile /var/run/redis_6379.pid
loglevel notice
logfile ""
databases 16
always-show-logo yes
save 900 1
save 300 10
save 60 10000
stop-writes-on-bgsave-error yes
rdbcompression yes
rdbchecksum yes
dbfilename dump.rdb
dir ./
slave-serve-stale-data yes
slave-read-only yes
repl-diskless-sync no
repl-diskless-sync-delay 5
repl-disable-tcp-nodelay no
slave-priority 100
lazyfree-lazy-eviction no
lazyfree-lazy-expire no
lazyfree-lazy-server-del no
slave-lazy-flush no
appendonly no
appendfilename "appendonly.aof"
appendfsync everysec
no-appendfsync-on-rewrite no
auto-aof-rewrite-percentage 100
auto-aof-rewrite-min-size 64mb
aof-load-truncated yes
aof-use-rdb-preamble no
lua-time-limit 5000
slowlog-log-slower-than 10000
slowlog-max-len 128
latency-monitor-threshold 0
notify-keyspace-events ""
hash-max-ziplist-entries 512
hash-max-ziplist-value 64
list-max-ziplist-size -2
list-compress-depth 0
set-max-intset-entries 512
zset-max-ziplist-entries 128
zset-max-ziplist-value 64
hll-sparse-max-bytes 3000
activerehashing yes
client-output-buffer-limit normal 0 0 0
client-output-buffer-limit slave 256mb 64mb 60
client-output-buffer-limit pubsub 32mb 8mb 60
client-query-buffer-limit 2gb
proto-max-bulk-len 4gb
hz 10
aof-rewrite-incremental-fsync yes
```

**.env_paperless**

```
USERMAP_UID=1000
USERMAP_GID=1000
PAPERLESS_SECRET_KEY=SECRET
PAPERLESS_TIME_ZONE=Europe/Berlin
PAPERLESS_OCR_LANGUAGE=deu
PAPERLESS_FILENAME_FORMAT={created_year}/{correspondent}/{title}
```

**.env_paperless-ng**

```
PAPERLESS_REDIS=redis://broker:6379
PAPERLESS_DBHOST=PRIVATE_IP
PAPERLESS_DBPORT=5432
PAPERLESS_DBNAME=paperless
PAPERLESS_DBUSER=PAPERLESS_DB_USER
PAPERLESS_DBPASS=PAPERLESS_DB_PASS
PAPERLESS_TIKA_ENABLED=1
PAPERLESS_TIKA_GOTENBERG_ENDPOINT=http://gotenberg:3000
PAPERLESS_TIKA_ENDPOINT=http://tika:9998
PAPERLESS_TASK_WORKERS=2
PAPERLESS_THREADS_PER_WORKER=2
PAPERLESS_CONVERT_MEMORY_LIMIT=2048
PAPERLESS_CONVERT_TMPDIR=/home/www-data/paperless-ng/tmp
```

**Comment by jonaswinkler** [comment]
Thanks for the info.

Memory usage stays about the same over here, even after adding 500 documents from scratch. I've just verified that again.

Anyway, I'll revert the change that causes this for now while I figure out a better solution. The only downside is that suggestions on the document detail page will load somewhat slower.

**Comment by jonaswinkler** [comment]
@sachilles

Could you please verify if this is still an issue with the latest dev version? Apart from this change, nothing major happened (https://paperless-ng.readthedocs.io/en/dev/changelog.html)

**Comment by jonaswinkler** [comment]
Also, even when using your redis config file, memory consumption stays the same when consuming many documents. Why do you need that?

**Comment by sachilles** [comment]
Hi,

first of all, thanks for your really fast reply. I checked the document consumption using the dev image. Ich can report that now, everything seems to work again. The maximum memory usage during consumption have been at 4 GB.

I used the redis configuration to get the "connection problems" between redis and paperless-ng under control (see also <https://stackoverflow.com/questions/64783283/connectionerror-error-104-while-writing-to-socket-connection-reset-by-peer>).

Currently I removed this configuration, because on the one hand it did not lead to success and on the other hand I wanted to know, if my problems are really solved by the "new" dev image. (And that's the way it is.)

Many thanks!
